### PR TITLE
OH2-226: add param wardCode on api opd search

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OH2-226.

### CODE REVIEW

File: src/main/java/org/isf/opd/rest/OpdController.java [hunk 1]
```diff
@@ -308,20 +310,24 @@ public ResponseEntity<Page<OpdDTO>> getOpdByDates(
...
-		Page<OpdDTO> opdPageable = new Page<>();
-		PagedResponse<Opd> opdsPaged = new PagedResponse<>();
+		LOGGER.debug("wardCode: {}", wardCode);
+		Page<OpdDTO> opdPageable = new Page<OpdDTO>();
+		PagedResponse<Opd> opdsPaged = new PagedResponse<Opd>();
 		List<Opd> opds = new ArrayList<>();
 		List<OpdDTO> opdDTOs =  new ArrayList<>();
-
+		Ward ward = null;
+        if (wardCode != null) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/opd/rest/OpdController.java:319]: This formatting looks incorrect

File: src/main/java/org/isf/opd/dto/OpdDTO.java [hunk 2]
```diff
@@ -117,7 +118,19 @@ public class OpdDTO {
...
-    public String getReason() {
+    @NotNull
+    @Schema(description = "ward")
+    private WardDTO ward;// ADDED: Arnaud
+    
+    public WardDTO getWard() {
+		return ward;
+	}
+
+	public void setWard(WardDTO ward) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/opd/dto/OpdDTO.java:129]: Strange indentation

File: src/main/java/org/isf/opd/dto/OpdDTO.java [hunk 3]
```diff
@@ -117,6 +118,10 @@ public class OpdDTO {
     @Schema(description = "prescription")
     private String prescription; // ADDED: Arnaud
     
+    @NotNull
+    @Schema(description = "ward")
+    private WardDTO ward;// ADDED: Arnaud
```
Comment (mwithi) [inline src/main/java/org/isf/opd/dto/OpdDTO.java:123]:
```suggestion
    private WardDTO ward;
```

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**